packages/legend-extension-external-language-morphir/src/MorphirGenerationPlugin.tsx

```diff
@@ -0,0 +1,123 @@
+  implements DSL_StudioPlugin_Extension
+{
+  networkClient: NetworkClient;
+
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+    this.networkClient = new NetworkClient();
+  }
+
+  install(pluginManager: StudioPluginManager): void {
+    pluginManager.registerStudioPlugin(this);
+  }
+
+  visualizeMorphir =
+    (fileNode: GenerationFile): (() => void) =>
+    (): void => {
+      this.networkClient.post(
+        `http://0.0.0.0:9901/insight`,
+        (fileNode as GenerationFile).content,
+      );
+      window.open('http://0.0.0.0:9901/insight');
+    };
+
+  visualizeBosque =
+    (
+      fileGenerationState: FileGenerationState,
+      fileNode: GenerationFile,
+    ): (() => void) =>
+    async (): Promise<void> => {
+      const code =
+        fileGenerationState.editorStore.graphManagerState.graphManager.graphToPureCode(
+          fileGenerationState.editorStore.graphManagerState.graph,
+        );
+      this.networkClient.post(`http://0.0.0.0:9900/lint`, {
+        ir: fileNode.content,
+        src: await code,
+      });
+      window.open('http://localhost:3050');
+    };
+
+  getVisualizeFileContentButtons(): ExtraFileGenerationtButton[] {
+    return [
+      (
+        fileGenerationState: FileGenerationState,
+        fileNode: GenerationFile,
+      ): React.ReactNode | undefined => {
+        if (this.isMorphirGenerationType(fileGenerationState)) {
+          return (
+            <div className="panel__header__title__content__with__margin generation-result-viewer__file__header-visualize-button">
+              <button
+                className="panel__content__form__section__list__new-item__add-btn btn btn--dark"
+                onClick={this.visualizeMorphir(fileNode)}
+                tabIndex={-1}
+              >
+                Visualize Generated IR
+              </button>
+            </div>
+          );
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getViewAdvancedFileInfoButtons(): ExtraFileGenerationtButton[] {
+    return [
+      (
+        fileGenerationState: FileGenerationState,
+        fileNode: GenerationFile,
+      ): React.ReactNode | undefined => {
+        if (this.isMorphirGenerationType(fileGenerationState)) {
+          return (
+            <div className="panel__header__title__content__with__margin generation-result-viewer__file__header-visualize-button">
+              <button
+                className="panel__content__form__section__list__new-item__add-btn btn btn--dark"
+                onClick={this.visualizeBosque(fileGenerationState, fileNode)}
+                tabIndex={-1}
+              >
+                View Bosque Feedback
+              </button>
+            </div>
+          );
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  isMorphirGenerationType(fileGenerationState: FileGenerationState): boolean {
+    return fileGenerationState.fileGeneration.type.toLowerCase() === `morphir`;
```

Comment:
Pleas turn this magic string into a constant and put it at the top of the file.

Reply:
Fixed!